bin/s2nc.c

```diff
@@ -607,6 +614,35 @@ int main(int argc, char *const *argv)
 
         GUARD_EXIT(s2n_connection_free_handshake(conn), "Error freeing handshake memory after negotiation");
 
+        if (send_file != NULL) {
+            printf("Sending file contents:\n%s\n", send_file);
+
+            unsigned long bytes_remaining_long = strlen(send_file);
+            GUARD_EXIT(bytes_remaining_long < INT_MAX, "send-file is too large");
+            ssize_t bytes_remaining = (ssize_t) bytes_remaining_long;
```

Comment:
Oh, is it for simplicity? So you don't have to do something like MIN(bytes_remaining, INT_MAX) for the call to s2n_send?

Reply:
It's because strlen returns an unsigned long, and s2n_send takes an ssize_t. So I didn't want to cast it down without checking first. Or is this fine to do because it's probably never going to happen?

Reply:
No, casting it is a bad idea. But the value you pass to s2n_send doesn't have to actually be the full size. You could send in batches, it just might complicate your code.

But like I said, not a huge deal, I can't imagine we'll want to send more than INT_MAX bytes for testing. And if we ever do we'll get a nice error message telling us to update s2nc to handle it.

Reply:
I see what you mean now! I think adding the MIN didn't make it too much more complex really, so I added it. Let me know if you'd rather have the guard, though.